src/util/piped_process.cpp

```diff
@@ -101,6 +101,19 @@
 
 #  define BUFSIZE 2048
 
+#ifdef _WIN32
+std::wstring process_windows_args(const std::vector<std::string> commandvec)
```

Comment:
:bulb: I would declare this as `static std::wstring process_windows_args(const std::vector<std::string> commandvec)`. Adding the `static` tells the compiler that this function has internal linkage only. Meaning that it can only be called from within this `.cpp` file. This informs the reader that this function is an internal implementation detail, not part of the external interface.